MathLib/LinAlg/SparsityPattern.h

```diff
@@ -0,0 +1,23 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATHLIB_LINALG_SPARSITYPATTERN_H
+#define MATHLIB_LINALG_SPARSITYPATTERN_H
+
+#include <vector>
+
+#include "ProcessLib/NumericsConfig.h"
+
+namespace MathLib
+{
+/// A vector telling how many nonzeros there are in each global matrix row.
+using SparsityPattern = std::vector<GlobalIndexType>;
```

Comment:
Providing exact diagonal and off-diagonal is the best choice for the computational efficiency. However, using d_nz and o_nz solely also do a better job, which has a sort of flexibility.